pkg/simulator/launch.go

```diff
@@ -1,72 +1,70 @@
 package simulator
 
 import (
+	"strings"
+
 	"github.com/controlplaneio/simulator-standalone/pkg/scenario"
 	"github.com/controlplaneio/simulator-standalone/pkg/ssh"
 	"github.com/pkg/errors"
-	"go.uber.org/zap"
-	"strings"
 )
 
 // Launch runs perturb.sh to setup a scenario with the supplied `id` assuming
 // the infrastructure has been created.  Returns an error if the infrastructure
 // is not ready or something goes wrong
-func Launch(logger *zap.SugaredLogger, tfDir, scenariosDir, bucketName, id, attackTag string) error {
-	logger.Debugf("Loading scenario manifest from %s", scenariosDir)
-	manifest, err := scenario.LoadManifest(scenariosDir)
+func (s *Simulator) Launch() error {
+	s.Logger.Debugf("Loading scenario manifest from %s", s.ScenariosDir)
+	manifest, err := scenario.LoadManifest(s.ScenariosDir)
 	if err != nil {
 		return errors.Wrap(err, "Error loading scenario manifest file")
 	}
 
-	logger.Debugf("Checking manifest contains %s", id)
-	if !manifest.Contains(id) {
-		return errors.Errorf("Scenario not found: %s", id)
+	s.Logger.Debugf("Checking manifest contains %s", s.ScenarioID)
+	if !manifest.Contains(s.ScenarioID) {
+		return errors.Errorf("Scenario not found: %s", s.ScenarioID)
 	}
 
-	logger.Debugf("Checking status of infrastructure")
-	tfo, err := Status(logger, tfDir, bucketName, attackTag)
-	if err != nil {
-		return err
-	}
+	s.Logger.Debugf("Checking status of infrastructure")
+	tfo, _ := s.Status()
+
 	if !tfo.IsUsable() {
 		return errors.Errorf("No infrastructure, please run simulator infra create")
 	}
-	logger.Debug(tfo)
+	s.Logger.Debug(tfo)
 
-	logger.Infof("Finding details of scenario %s", id)
-	s := manifest.Find(id)
-	logger.Debug(s)
+	s.Logger.Infof("Finding details of scenario %s", s.ScenarioID)
+	sID := manifest.Find(s.ScenarioID)
```

Comment:
can we call this `foundScenario` or something like that rather than `sID`?   It makes it a bit clearer what this is